src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -628,10 +654,69 @@ MetadataCreateCommands(void)
 												  shardCreateCommandList);
 	}
 
+	/* As the last step, propagate the pg_dist_object entities */
+	metadataSnapshotCommandList = list_concat(
+		metadataSnapshotCommandList,
+		DistributedObjectSyncCommandList());
+
 	return metadataSnapshotCommandList;
 }
 
 
+/*
+ * DistributedObjectSyncCommandList returns the necessary commands to create pg_dist_object entries
+ * on the new node.
+ */
+static List *
+DistributedObjectSyncCommandList(void)
+{
+	List *commandList = NIL;
+	HeapTuple pgDistObjectTup = NULL;
+	Relation pgDistObjectRel = table_open(DistObjectRelationId(), AccessShareLock);
+	TupleDesc pgDistObjectDesc = RelationGetDescr(pgDistObjectRel);
+
+	List *objectAddresses = NIL;
+	List *distributionArgumentIndexes = NIL;
+	List *colocationIds = NIL;
+
+	SysScanDesc pgDistObjectScan = systable_beginscan(pgDistObjectRel, InvalidOid, false,
+													  NULL, 0, NULL);
+	while (HeapTupleIsValid(pgDistObjectTup = systable_getnext(pgDistObjectScan)))
```

Comment:
there might a bug here, which I couldn't spot it yet.

I created 100k objects see [here](https://gist.githubusercontent.com/onderkalaci/01b83abfc849995d406a292084391aa3/raw/4a538c62ea50bc1243d4275353a7ba925a026f89/100k_objects.sql).

Then, followed some of the sequence regression tests:
```SQL
-- check that there's not problem with group ID cache
CREATE TABLE seq_test_4 (x int, y int);
SELECT create_distributed_table('seq_test_4','x');
CREATE SEQUENCE seq_4;
ALTER TABLE seq_test_4 ADD COLUMN a bigint DEFAULT nextval('seq_4');
SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
DROP SEQUENCE seq_4 CASCADE;
TRUNCATE seq_test_4;
CREATE SEQUENCE seq_4;
ALTER TABLE seq_test_4 ADD COLUMN b bigint DEFAULT nextval('seq_4');
```

And, realized that we keep sending a huge block of sequence objects when metadata synec, where we should have been sending the other objects in the file:
```SQL
('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL), ('sequence', ARRAY['public', 'seq_4']::text[], ARRAY[]::text[],NULL, NULL)) SELECT citus_internal_add_object_metadata(typetext, objnames, objargs, distargumentindex::int, colocationid::int) FROM distributed_object_data;
```

Then, see that we miss all the
```SQL
select count(*) from citus.pg_dist_object;;
 count  
--------
 100504
(1 row)


SELECT run_command_on_workers($$select count(*) from citus.pg_dist_object;$$);
  run_command_on_workers   
---------------------------
 (localhost,9700,t,1)
 (localhost,9701,t,100503)
 (localhost,9702,t,100503)
 (localhost,9703,t,100503)
 (localhost,9704,t,100503)
 (localhost,9705,t,100503)
 (localhost,9706,t,100503)
(7 rows)
```

Reply:
It might be because of adding the pointer of a local variable `ObjectAddress address` to the list and passing it to another function.

We should follow something like `FilterNameListForDistributedCollations`:
```C
			ObjectAddress *address = palloc0(sizeof(ObjectAddress));
			*address = collAddress;
			*objectAddresses = lappend(*objectAddresses, address);
			result = lappend(result, collName);
```

Reply:
Yes, I've also realized that and just pushed the related change.